gui/src/app/pyodide/pyodideWorker/pyodideWorkerTypes.ts

```diff
@@ -0,0 +1,69 @@
+export type PyodideWorkerMode = "data.py" | "analysis.py";
+
+export type MessageToPyodideWorker =
+  | {
+      type: "setPyodideWorkerMode";
+      mode: PyodideWorkerMode;
+    }
+  | {
+      type: "run";
+      code: string;
+      globalData: { [key: string]: any };
+    };
+
+export const isMessageToPyodideWorker = (
+  x: any,
+): x is MessageToPyodideWorker => {
+  if (!x) return false;
+  if (typeof x !== "object") return false;
+  if (x.type === "setPyodideWorkerMode")
+    return ["data.py", "analysis.py"].includes(x.mode);
+  if (x.type === "run") {
+    if (x.code === undefined) return false;
+    if (x.globalData === undefined) return false;
+    return true;
+  }
+  return false;
+};
+
+export type MessageFromPyodideWorker =
+  | {
+      type: "stdout" | "stderr";
+      data: string;
+    }
+  | {
+      type: "setStatus";
+      status: PydodideWorkerStatus;
+    }
+  | {
+      type: "setData"; // for data.py mode
+      data: any;
+    }
+  | {
+      type: "addImage"; // for analysis.py mode
+      image: any;
+    };
+
+export const isMessageFromPyodideWorker = (
+  x: any,
+): x is MessageFromPyodideWorker => {
+  if (!x) return false;
+  if (typeof x !== "object") return false;
+  if (x.type === "stdout") return x.data !== undefined;
+  if (x.type === "stderr") return x.data !== undefined;
+  if (x.type === "setStatus") return isPydodideWorkerStatus(x.status);
+  if (x.type === "setData") return x.data !== undefined;
+  if (x.type === "addImage") return x.image !== undefined;
+  return false;
+};
+
+export type PydodideWorkerStatus =
+  | "idle"
+  | "loading"
+  | "running"
+  | "completed"
+  | "failed";
+
+export const isPydodideWorkerStatus = (x: any): x is PydodideWorkerStatus => {
+  return ["idle", "loading", "running", "completed", "failed"].includes(x);
+};
```

Comment:
It has its own drawbacks, but defining the status as an enum allows you to pull the keys/values of the enum as a list, which might make the `includes` check here more intuitive. (That said we probably aren't expanding the number of statuses any time soon, so maybe not worth worrying about?)